lib/matplotlib/axes/_axes.py

```diff
@@ -3398,112 +3398,54 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         barcols = []
         caplines = []
 
-        # arrays fine here, they are booleans and hence not units
-        lolims = np.broadcast_to(lolims, len(x)).astype(bool)
-        uplims = np.broadcast_to(uplims, len(x)).astype(bool)
-        xlolims = np.broadcast_to(xlolims, len(x)).astype(bool)
-        xuplims = np.broadcast_to(xuplims, len(x)).astype(bool)
-
         # Vectorized fancy-indexer.
         def apply_mask(arrays, mask): return [array[mask] for array in arrays]
 
-        def extract_err(name, err, data, lolims, uplims):
-            """
-            Private function to compute error bars.
-
-            Parameters
-            ----------
-            name : {'x', 'y'}
-                Name used in the error message.
-            err : array-like
-                xerr or yerr from errorbar().
-            data : array-like
-                x or y from errorbar().
-            lolims : array-like
-                Error is only applied on **upper** side when this is True.  See
-                the note in the main docstring about this parameter's name.
-            uplims : array-like
-                Error is only applied on **lower** side when this is True.  See
-                the note in the main docstring about this parameter's name.
-            """
-            try:
-                low, high = np.broadcast_to(err, (2, len(data)))
+        # dep: dependent dataset, indep: independent dataset
+        for name, dep, err, lolims, uplims, indep, lines, mk, lomk, himk in [
```

Comment:
```suggestion
        for (name, dep, err, lolims, uplims, indep, lines,
             marker, lomarker, himarker) in [
```
This is already cumbersome to follow, saving characters makes it worse.
Breaking the unpacked tuple over two lines is actually better here because we can break at the same position as we break the values.

Optional: Replace `lines` by `lines_func`. While the underlying function is `h/vlines`, `a free variables `lines` rather reminds of a collection of Line objects.

Optional: It may be worth considering keeping `x` and `y` instead of `dep and `indep` and adding a comment:

> For better readability the naming of the parameters follows the case of
> x-errors. However the second run uses the same logic for y-errors.

Reply:
yes to the first two points.  I *think* I prefer keeping `dep`/`indep` (in particular, overwriting the original `x`/`y` seems a bit messy, and introducing other variables not great either), but don't feel very strongly about it either.